vault/core.go

```diff
@@ -1692,6 +1692,18 @@ func (c *Core) postUnseal(ctx context.Context, ctxCancelFunc context.CancelFunc,
 		return err
 	}
 
+	// Automatically re-encrypt the keys used for auto unsealing when the
+	// seal's encryption key changes. The regular rotation of cryptographic
+	// keys is a NIST recommendation. Access to prior keys for decryption
+	// is normally supported for a configurable time period. Re-encrypting
+	// the keys used for auto unsealing ensures Vault and its data will
+	// continue to be accessible even after prior seal keys are destroyed.
+	if seal, ok := c.seal.(*autoSeal); ok {
+		if err := seal.UpgradeKeys(c.activeContext); err != nil {
+			c.logger.Warn("post-unseal upgrade seal keys failed", "error", err)
```

Comment:
```suggestion
			c.logger.Error("post-unseal upgrade seal keys failed", "error", err)
```

Reply:
Per our 1-on-1 discussion, leaving this to log at the Warn level. Overall reasoning: an error from `seal.UpgradeKeys` could be a transient error which is why `postUnseal` continues.